src/Parser/Mage/Fire/Modules/Features/Abilities.js

```diff
@@ -33,6 +46,7 @@ class Abilities extends CoreAbilities {
       {
         spell: SPELLS.FIRE_BLAST,
         category: Abilities.SPELL_CATEGORIES.ROTATIONAL,
+        isOnGCD: false,
         cooldown: haste => 12 / (1 + haste),
         charges: 2,
```

Comment:
instead of having 2 entries for Fireblast you could set the number of charges based on the current enabled logic.

Reply:
how would i do that?? Every other instance of this I have seen has 2 entries. If you point me in the direction of a spec that has what you are referring to then ill change it.

Reply:
`charges: combatant.hasTalent(SPELLS.FLAME_ON_TALENT.id) || combatant.hasFinger(ITEMS.SOUL_OF_THE_ARCHMAGE.id) ? 3 : 2;`

Reply:
ahh, didnt think about that. Will do.